kyo-prelude/shared/src/main/scala/kyo/kernel/Boundary.scala

```diff
@@ -1,9 +1,24 @@
 package kyo.kernel
 
 import internal.*
+import kyo.Ansi.*
+import scala.annotation.implicitNotFound
 import scala.annotation.nowarn
 import scala.quoted.*
 
+@implicitNotFound("""
+Could not create an Async boundary for effects: ${Ctx}
+
+Note: If you're seeing this error with nested Async operations (like Async.run(Async.run(v))),
+this is due to a current limitation. As a workaround, you can break them into separate statements:
+
+Instead of:
+  Async.run(Async.run(v))
+
+Use:
+  val x = Async.run(v)
+  Async.run(x)
```

Comment:
The macro doesn't even get invoked and the compiler complains of not being able to locate the implicit for the case  where boundaries are nested.